### PR TITLE
docs: more runtime details

### PR DESCRIPTION
**Pull request by mattklein123**
fixes https://github.com/lyft/envoy/issues/612

### CODE REVIEW

**Comment by mattklein123** [comment]
@lyft/network-team @rshriram

File: docs/configuration/overview/runtime.rst [hunk 1]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+* *subdirectory*: ``envoy``
+* *override_subdirectory*: ```envoy_override``
+
+Each '.' in a runtime key indicates a new directory in the hierarchy, rooted at *symlink_root* +
+*subdirectory*. For example, the *health_check.min_interval* path would have the following full
+file system path:
+
+``/srv/runtime/current/envoy/health_check/min_interval``
+
+The terminal portion of the path is a file that generally contains a numeric value unless otherwise
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: Break into smaller sentences? Like
The terminal portion of a path is the file. The contents of the file constitute the runtime value. When reading numeric values from a file, spaces and new lines will be ignored.

File: docs/configuration/overview/runtime.rst [hunk 2]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+documented. Envoy will strip white space as needed when reading the file.
+
+The *override_subdirectory* is used along with the :option:`--service-cluster` CLI option. Assume
+that :option:`--service-cluster` has been set to ``my-clustser``. Envoy will first look for the
+*health_check.min_interval* key in the following full file system path:
+
+``/srv/runtime/current/envoy_override/my-cluster/health_check/min_interval``
+
+If found, the value will override any value found in the primary lookup path. This allows runtime to
+be customized for individual clusters as needed on top of a global base.
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: This allows the user to customize the runtime values for individual clusters on top of global defaults.

File: docs/configuration/overview/runtime.rst [hunk 3]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+
+Assume the following configuration settings for runtime:
+
+* *symlink_root*: ``/srv/runtime/current``
+* *subdirectory*: ``envoy``
+* *override_subdirectory*: ```envoy_override``
+
+Each '.' in a runtime key indicates a new directory in the hierarchy, rooted at *symlink_root* +
+*subdirectory*. For example, the *health_check.min_interval* path would have the following full
+file system path:
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: would have the following path (using the symbolic link)

File: docs/configuration/overview/runtime.rst [hunk 4]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+  configuration. See below for exactly how the override directory is used.
+
+File system layout
+------------------
+
+Various sections of the configuration guide describe the runtime settings that are available.
+For example, :ref:`here <config_cluster_manager_cluster_runtime>` are the runtime settings for
+upstream clusters.
+
+Assume the following configuration settings for runtime:
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: Assume that the folder /etc/config/v1/envoy points to the actual filesystem where global envoy runtime configurations are stored. The following would be a typical configuration setting for the runtime:
...

where, /srv/runtime/current would be a symlink to /etc/config/v1.

File: docs/configuration/overview/runtime.rst [hunk 5]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+The *override_subdirectory* is used along with the :option:`--service-cluster` CLI option. Assume
+that :option:`--service-cluster` has been set to ``my-clustser``. Envoy will first look for the
+*health_check.min_interval* key in the following full file system path:
+
+``/srv/runtime/current/envoy_override/my-cluster/health_check/min_interval``
+
+If found, the value will override any value found in the primary lookup path. This allows runtime to
+be customized for individual clusters as needed on top of a global base.
+
+Symbolic link management
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: Rename to Updating runtime values

File: docs/configuration/overview/runtime.rst [hunk 6]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+------------------------
+
+Envoy uses a fairly common pattern for atomic file system updates. Envoy assumes that an entire
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: There are two steps to update any runtime value. First, create a hard copy of the entire runtime tree and update the desired runtime values. Second, atomically swap the symlink root from the old tree to the new runtime tree, using the equivalent of the following command:

File: docs/configuration/overview/runtime.rst [hunk 7]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+runtime tree is written to disk, and the symbolic link pointed to by *symlink_root* is swapped
+using the equivalent of the following command:
+
+..  code-block:: console
+
+  ln -s new current_tmp && mv -Tf current_tmp current
```
Comment (rshriram) [inline docs/configuration/overview/runtime.rst]: ```/srv/runtime:~$ ln -s /etc/config/v2 new && mv -Tf new current```

File: docs/configuration/overview/runtime.rst [hunk 8]
```diff
@@ -33,7 +33,51 @@ override_subdirectory
...
+*subdirectory*. For example, the *health_check.min_interval* path would have the following full
+file system path:
+
+``/srv/runtime/current/envoy/health_check/min_interval``
+
+The terminal portion of the path is a file that generally contains a numeric value unless otherwise
+documented. Envoy will strip white space as needed when reading the file.
+
+The *override_subdirectory* is used along with the :option:`--service-cluster` CLI option. Assume
+that :option:`--service-cluster` has been set to ``my-clustser``. Envoy will first look for the
```
Comment (junr03) [inline docs/configuration/overview/runtime.rst]: s/clustser/cluster

**Comment by mattklein123** [comment]
@rshriram @junr03 updated

**Pull request merged**